### ISSUE TITLE
Error when getting filtered paginated results

### ISSUE THREAD

**Issue by citaret** [linode/python-linode-api#42]
It is related to issue https://github.com/linode/python-linode-api/issues/34

The result is OK for kernels list, but failed for filtered kernels list.

```
In [3]: import linode

In [4]: client = linode.LinodeClient('unauthenticated')

In [5]: kernels = client.linode.get_kernels()

In [6]: kernels_filtered = client.linode.get_kernels(linode.Kernel.x64 == True)

In [7]: kernels
Out[7]: PaginatedList (180 items)

In [8]: kernels_filtered
Out[8]: PaginatedList (87 items)

In [9]: kernels[25]
Out[9]: Kernel: linode/4.0.5-x86-linode76

In [10]: kernels_filtered[25]
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-10-16ba93c9f3c4> in <module>()
----> 1 kernels_filtered[25]

/usr/local/lib/python3.5/dist-packages/linode/util/paginated_list.py in __getitem__(self, index)
     62 
     63         if not self.lists[target_page]:
---> 64             self._load_page(target_page)
     65 
     66         return self.lists[target_page][normalized_index]

/usr/local/lib/python3.5/dist-packages/linode/util/paginated_list.py in _load_page(self, page_number)
     39 
     40         if j['total_pages'] != self.max_pages or j['total_results'] != len(self):
---> 41             raise RuntimeError('List {} has changed since creation!'.format(self))
     42 
     43         l = mappings.make_list(j[self.key], self.client, parent_id=self.objects_parent_id, cls=self.list_cls)

RuntimeError: List PaginatedList (87 items) has changed since creation!
```

Thanks.

**Comment by citaret** [comment]
Have a quick view of the code, seems `PaginatedList` does not support filtered result yet.

**Comment by Dorthu** [comment]
I think you're right, PaginatedList doesn't work with filtering.  I'll fix this in the next release.

**Comment by Dorthu** [comment]
This has been addressed, and will be in version 4.1.3b0.  Thanks again for reporting it!

**Comment by Dorthu** [comment]
This fix has been released in v4.1.3b1
